xchart/src/main/java/org/knowm/xchart/internal/chartpart/PlotContentLineChart.java

```diff
@@ -133,20 +143,23 @@ public void paint(Graphics2D g) {
       }
       Path2D.Double path = null;
 
-      while (xItr.hasNext()) {
-
-        double x = 0.0;
-        if (getChartPainter().getAxisPair().getXAxis().getAxisType() == AxisType.Number) {
-          x = ((Number) xItr.next()).doubleValue();
-          // System.out.println(x);
-        }
-        else if (getChartPainter().getAxisPair().getXAxis().getAxisType() == AxisType.Date) {
-          x = ((Date) xItr.next()).getTime();
-          // System.out.println(x);
-        }
-
-        if (getChartPainter().getStyleManager().isXAxisLogarithmic()) {
-          x = Math.log10(x);
+      double x = -0.5;
```

Comment:
Setting this to `-0.5` instead of `-1.0` is a hack to try and get the line to match to the x-axis.  Though this does move the first point to match the x axis you can see in the example graph image that it does not take many points for that offset to become incorrect.  And the end of the graph is a full position beyond what it should be.

I could use some direction as to how to get these to line up perfectly.